phases/core_builder.go

```diff
@@ -124,10 +124,14 @@ func compileCore(ctx *types.Context, buildPath string, buildCachePath string, bu
 
 	// archive core.a
 	if targetArchivedCore != "" {
+		err := builder_utils.CopyFile(archiveFile, targetArchivedCore)
 		if ctx.Verbose {
-			logger.Println(constants.LOG_LEVEL_INFO, constants.MSG_ARCHIVING_CORE_CACHE, targetArchivedCore)
+			if err == nil {
+				logger.Println(constants.LOG_LEVEL_INFO, constants.MSG_ARCHIVING_CORE_CACHE, targetArchivedCore)
+			} else {
+				logger.Println(constants.LOG_LEVEL_INFO, constants.MSG_CORE_CACHE_UNAVAILABLE, ctx.ActualPlatform.PlatformId)
```

Comment:
This blames *any* error while copying the file on the core maintainers. I suppose the intended case is where the core does not use the proper .a filename, so the copy cannot find it? If so, then perhaps a separate check should be made to see if the .a filename exists and give this error if it does not? If the file *does* exist, but the copy fails, a new error message should probably be shown?

Reply:
I'd also add `arduino.ar` as a possible target but this clashes badly with specifications at http://goo.gl/QdCUjo .
I think that adding this check will make life terrible for maintainers of "incompatible" cores for a bit but will enormously speedup the adoption of proper naming (esp8266 just merged it in mainline https://github.com/esp8266/Arduino/commit/85e68093e94b4e5c5e711888b8b5facee1bd4f9f#diff-d638a47d1ca0e2a5585fb34c7d7fb0f9L95 )

Reply:
Terrible in what way? Because users will start pestering them? Or because their incompatible cores will break or become slower (I can't see why or how)?

Reply:
The first one :slightly_smiling_face: